tests/Unit/Controllers/Service/Admin/VoucherControllerTest.php

```diff
@@ -0,0 +1,117 @@
+                'end' => 'The end field is required.',
+                'sponsor_id' => 'The sponsor id field is required.',
+            ])
+        ;
+    }
+
+    public function testStoreBatchStartEndSwapped() {
+        $this->actingAs($this->admin_user, 'admin')
+            ->post(route('admin.vouchers.storebatch'), [
+                'sponsor_id' => $this->market->sponsor_id,
+                'start' => '10',
+                'end' => '-1',
+            ])
+            ->assertStatus(302)
+            ->assertSessionMissing('notification')
+            ->assertSessionHasErrors([
+                'end' => 'The end must be greater than or equal to start',
+            ])
+        ;
+    }
+
+    public function testStoreBatchInvalidSponsor() {
+        $this->actingAs($this->admin_user, 'admin')
+            ->post(route('admin.vouchers.storebatch'), [
+                'sponsor_id' => $this->market->sponsor_id + 1,
+            ])
+            ->assertStatus(302)
+            ->assertSessionMissing('notification')
+            ->assertSessionHasErrors([
+                'sponsor_id' => 'The sponsor id must be a valid selection',
+            ])
+        ;
+    }
+
+    public function testStoreBatchSuccessMsg() {
+        $shortcode = $this->market->sponsor_shortcode;
+        $start = '1';
+        $end = '10';
+        $notification_msg = trans('service.messages.vouchers_create_success',[
+            'shortcode' => $shortcode,
+            'start' => $start,
+            'end' => $end,
+        ]);
+        $this->actingAs($this->admin_user, 'admin')
+            ->post(route('admin.vouchers.storebatch'), [
+                'sponsor_id' => $this->market->sponsor_id,
+                'start' => $start,
+                'end' => $end,
+            ])
+            ->assertStatus(302)
+            ->assertSessionHas('notification', $notification_msg)
+        ;
+    }
+
+    public function testStoreBatch() {
+        $shortcode = $this->market->sponsor_shortcode;
+        $this->actingAs($this->admin_user, 'admin')
+            ->post(route('admin.vouchers.storebatch'), [
+                'sponsor_id' => $this->market->sponsor_id,
+                'start' => '50',
+                'end' => '61',
+            ])
+            ->assertStatus(302)
+        ;
+
+        $vouchers = Voucher::findMany(range(0, 11));
+
+        // Assert that eleven vouchers have been made and are in the expected state and land within the expected range.
+        $this->assertCount(11, $vouchers);
+        foreach($vouchers as $voucher) {
+            $this->assertEquals('printed', $voucher->currentstate);
+
+            // Assert that the voucher code starts with the Sponsor shortcode.
+            $this->assertStringStartsWith($shortcode, $voucher->code);
+
+            // Assert that the voucher code lands between 50 and 60 (our generated range).
+            $this->assertRegExp('/,*[5-6][0-9]/', $voucher->code);
```

Comment:
I'll look into narrowing after I've got the fix in for the number of generated vouchers in that other ticket.